File: examples/03-interactions/go/schedule_interval.go
package interactions

import (
	"context"
	"time"

	enumspb "go.temporal.io/api/enums/v1"
	"go.temporal.io/sdk/client"
)

// A Schedule is a server-managed cron replacement. The interval spec fires every
// fixed period; jitter spreads the fire time to avoid thundering herds, and the
// overlap policy decides what happens when a run is still going as the next fires.
func CreateHourlySchedule(ctx context.Context, c client.Client) error {
	_, err := c.ScheduleClient().Create(ctx, client.ScheduleOptions{
		ID: "hourly-orders",
		Spec: client.ScheduleSpec{
			Intervals: []client.ScheduleIntervalSpec{{Every: time.Hour}},
			Jitter:    5 * time.Minute,
		},
		Action: &client.ScheduleWorkflowAction{
			ID:        "hourly-orders-wf",
			Workflow:  OrdersWorkflow,
			Args:      []any{"hourly"},
			TaskQueue: "orders",
		},
		Overlap: enumspb.SCHEDULE_OVERLAP_POLICY_BUFFER_ONE,
	})
	return err
}

File: examples/03-interactions/python/schedule_interval.py
from datetime import timedelta

from temporalio.client import (
    Client,
    Schedule,
    ScheduleActionStartWorkflow,
    ScheduleIntervalSpec,
    ScheduleOverlapPolicy,
    SchedulePolicy,
    ScheduleSpec,
)


# A Schedule is a server-managed cron replacement. The interval spec fires every
# fixed period; jitter spreads the fire time to avoid thundering herds, and the
# overlap policy decides what happens when a run is still going as the next fires.
async def create_hourly_schedule(client: Client) -> None:
    await client.create_schedule(
        "hourly-orders",
        Schedule(
            action=ScheduleActionStartWorkflow(
                OrdersWorkflow.run,
                "hourly",
                id="hourly-orders-wf",
                task_queue="orders",
            ),
            spec=ScheduleSpec(
                intervals=[ScheduleIntervalSpec(every=timedelta(hours=1))],
                jitter=timedelta(minutes=5),
            ),
            policy=SchedulePolicy(overlap=ScheduleOverlapPolicy.BUFFER_ONE),
        ),
    )
